app/charts/shared/chart-dimensions.tsx

```diff
@@ -211,3 +226,60 @@ export const useAxisLabelHeightOffset = ({
     labelWidth,
   };
 };
+
+export const getLongestXLabel = ({
+  xScale,
+  getXLabel,
+  xTimeUnit,
+  formatDate,
+  fontSize,
+}: {
+  xScale: ScaleBand<string>;
+  getXLabel: (d: any) => string;
+  xTimeUnit?: string;
+  formatDate?: (d: any, timeUnit: string) => string;
```

Comment:
@noahonyejese I think we can use a correct type here instead of any, isn't d supposed to be `Observation` and `ObservationValue`?